pubsub/kafkapubsub: flaky data race in tests

```
=================
WARNING: DATA RACE
Write at 0x00c0005de4d0 by goroutine 98:
  gocloud.dev/pubsub/kafkapubsub.(*subscription).Setup()
      /home/travis/gopath/src/gocloud.dev/pubsub/kafkapubsub/kafka.go:416 +0xde
  github.com/Shopify/sarama.newConsumerGroupSession()
      /home/travis/gopath/pkg/mod/github.com/!shopify/sarama@v1.19.0/consumer_group.go:505 +0x688
  github.com/Shopify/sarama.(*consumerGroup).newSession()
      /home/travis/gopath/pkg/mod/github.com/!shopify/sarama@v1.19.0/consumer_group.go:270 +0x6f5
  github.com/Shopify/sarama.(*consumerGroup).Consume()
      /home/travis/gopath/pkg/mod/github.com/!shopify/sarama@v1.19.0/consumer_group.go:172 +0x29f
  gocloud.dev/pubsub/kafkapubsub.openSubscription.func1()
      /home/travis/gopath/src/gocloud.dev/pubsub/kafkapubsub/kafka.go:390 +0xb4
Previous read at 0x00c0005de4d0 by goroutine 113:
  gocloud.dev/pubsub/kafkapubsub.(*subscription).As()
      /home/travis/gopath/src/gocloud.dev/pubsub/kafkapubsub/kafka.go:589 +0x18f
  gocloud.dev/pubsub.(*Subscription).As()
      /home/travis/gopath/src/gocloud.dev/pubsub/pubsub.go:742 +0x80
  gocloud.dev/pubsub/kafkapubsub.asTest.SubscriptionCheck()
      /home/travis/gopath/src/gocloud.dev/pubsub/kafkapubsub/kafka_test.go:159 +0x10e
  gocloud.dev/pubsub/kafkapubsub.(*asTest).SubscriptionCheck()
      <autogenerated>:1 +0x49
  gocloud.dev/pubsub/drivertest.testAs()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:848 +0x246
  gocloud.dev/pubsub/drivertest.RunConformanceTests.func2.1()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:188 +0x9e
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.10.linux.amd64/src/testing/testing.go:827 +0x162
Goroutine 98 (running) created at:
  gocloud.dev/pubsub/kafkapubsub.openSubscription()
      /home/travis/gopath/src/gocloud.dev/pubsub/kafkapubsub/kafka.go:389 +0x3fc
  gocloud.dev/pubsub/kafkapubsub.(*harness).CreateSubscription()
      /home/travis/gopath/src/gocloud.dev/pubsub/kafkapubsub/kafka_test.go:119 +0x340
  gocloud.dev/pubsub/drivertest.makePair()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:810 +0x198
  gocloud.dev/pubsub/drivertest.testAs()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:838 +0x15a
  gocloud.dev/pubsub/drivertest.RunConformanceTests.func2.1()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:188 +0x9e
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.10.linux.amd64/src/testing/testing.go:827 +0x162
Goroutine 113 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.11.10.linux.amd64/src/testing/testing.go:878 +0x659
  gocloud.dev/pubsub/drivertest.RunConformanceTests.func2()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:188 +0x26d
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.10.linux.amd64/src/testing/testing.go:827 +0x162
==================
--- FAIL: TestConformance (48.29s)
    --- FAIL: TestConformance/TestAs (9.76s)
        --- FAIL: TestConformance/TestAs/kafka (4.88s)
            testing.go:771: race detected during execution of test
        testing.go:771: race detected during execution of test
    testing.go:771: race detected during execution of test
FAIL
coverage: 8.3% of statements in ./...
FAIL	gocloud.dev/pubsub/kafkapubsub	53.377s
```

--- a/pubsub/kafkapubsub/kafka.go
+++ b/pubsub/kafkapubsub/kafka.go
@@ -585,7 +585,7 @@ func (s *subscription) As(i interface{}) bool {
 	}
 	if p, ok := i.(*sarama.ConsumerGroupSession); ok {
 		s.mu.Lock()
-		s.mu.Unlock()
+		defer s.mu.Unlock()
 		*p = s.sess
 		return true
 	}

